src/ngAnimate/animateQueue.js

```diff
@@ -0,0 +1,558 @@
+              activeAnimationsLookup.remove(child);
+            }
+            break;
+        }
+      });
+    }
+
+    function clearElementAnimationState(element) {
+      element = element.length ? element[0] : element;
+      element.removeAttribute(NG_ANIMATE_ATTR_NAME);
+      activeAnimationsLookup.remove(element);
+    }
+
+    function isMatchingElement(a,b) {
+      a = a.length ? a[0] : a;
+      b = b.length ? b[0] : b;
+      return a === b;
+    }
+
+    function closeParentClassBasedAnimations(startingElement) {
+      var parentNode = startingElement[0];
+      do {
+        if (!parentNode || parentNode.nodeType !== ELEMENT_NODE) break;
+
+        var details = activeAnimationsLookup.get(parentNode);
+        if (details) {
+          examineParentAnimation(parentNode, details);
+        }
+
+        parentNode = parentNode.parentNode;
+      } while (true);
+
+      // since animations are detected from CSS classes, we need to flush all parent
+      // class-based animations so that the parent classes are all present for child
+      // animations to properly function (otherwise any CSS selectors may not work)
+      function examineParentAnimation(node, animationDetails) {
+        // enter/leave/move always have priority
+        if (animationDetails.structural) return;
+
+        if (animationDetails.state === RUNNING_STATE) {
+          animationDetails.runner.end();
+        }
+        clearElementAnimationState(node);
+      }
+    }
+
+    function areAnimationsAllowed(element, parent, event) {
+      var bodyElement = jqLite($document[0].body);
+      var bodyElementDetected = false;
+      var rootElementDetected = false;
+      var parentAnimationDetected = false;
+      var animateChildren;
+
+      while (parent && parent.length) {
+        var parentNode = parent[0];
+        if (parentNode.nodeType !== ELEMENT_NODE) {
+          // no point in inspecting the #document element
+          break;
+        }
+
+        var details = activeAnimationsLookup.get(parentNode) || {};
+        // either an enter, leave or move animation will commence
+        // therefore we can't allow any animations to take place
+        // but if a parent animation is class-based then that's ok
+        if (!parentAnimationDetected) {
+          parentAnimationDetected = details.structural || disabledElementsLookup.get(parentNode);
+        }
+
+        if (isUndefined(animateChildren) || animateChildren === true) {
+          var value = parent.data(NG_ANIMATE_CHILDREN_DATA);
+          if (isDefined(value)) {
+            animateChildren = value;
+          }
+        }
+
+        if (!rootElementDetected) {
+          // angular doesn't want to attempt to animate elements outside of the application
+          // therefore we need to ensure that the rootElement is an ancestor of the current element
+          rootElementDetected = isMatchingElement(parent, $rootElement);
+        }
+
+        if (!bodyElementDetected) {
```

Comment:
Because it will only check the body element once the rootElement is detected and when it reaches `<html>` then it's too late. We could add a special case for `parentNode.nodeType === 9` to check if it reaches the `document` object.